src/components/Modals/ReplaceCourseModal.vue

```diff
@@ -0,0 +1,294 @@
+      } else {
+        // closes the modal if the course exists exactly once
+        this.closeCurrentModal();
+      }
+    },
+    selectCourse(result: CornellCourseRosterCourse) {
+      this.selectedCourse = result;
+      this.$emit('select-course', this.selectedCourse);
+      this.getReqsRelatedToCourse(result);
+      this.handleAdd();
+    },
+    closeCurrentModal() {
+      this.$emit('close-replace-course-modal');
+    },
+    getReqsRelatedToCourse(selectedCourse: CornellCourseRosterCourse) {
+      const {
+        relatedRequirements,
+        selfCheckRequirements,
+        automaticallyFulfilledRequirements,
+      } = getRelatedUnfulfilledRequirements(
+        selectedCourse,
+        store.state.groupedRequirementFulfillmentReport,
+        store.state.onboardingData,
+        store.state.toggleableRequirementChoices,
+        store.state.overriddenFulfillmentChoices,
+        store.state.userRequirementsMap
+      );
+      const automaticallyFulfilledRequirementIds = new Set(
+        automaticallyFulfilledRequirements.map(({ id }) => id)
+      );
+
+      const reqFilter = (req: RequirementWithIDSourceType) =>
+        !automaticallyFulfilledRequirementIds.has(req.id);
+
+      this.automaticallyFulfilledRequirements = automaticallyFulfilledRequirements.map(
+        ({ name }) => name
+      );
+      this.relatedRequirements = relatedRequirements.filter(reqFilter);
+      this.selfCheckRequirements = selfCheckRequirements.filter(reqFilter);
+      if (this.relatedRequirements.length > 0) {
+        this.selectedRequirementID = this.relatedRequirements[0].id;
+      } else {
+        this.selectedRequirementID = '';
+      }
+    },
+    addItem() {
+      if (this.editMode) {
+        this.editMode = false;
+      } else {
+        this.addCourse();
+      }
+    },
+    addCourse() {
+      if (this.selectedCourse == null) return;
+      this.$emit('add-course', this.selectedCourse, {
+        optOut: getRelatedRequirementIdsForCourseOptOut(
+          this.selectedCourse.crseId,
+          this.selectedRequirementID,
+          store.state.groupedRequirementFulfillmentReport,
+          store.state.toggleableRequirementChoices,
+          store.state.userRequirementsMap
+        ),
+        // Only include the selected requirement from opt-in.
+        acknowledgedCheckerWarningOptIn: this.selfCheckRequirements
+          .filter(it => it.id === this.selectedRequirementID)
+          .map(it => it.id),
+        arbitraryOptIn: {},
+      });
+      this.closeCurrentModal();
+    },
+    onSelectedChange(selected: string) {
+      this.selectedRequirementID = selected;
+    },
+    backOrCancel() {
+      if (this.leftButtonText === leftButtonState.Back) {
+        if (this.editMode) {
+          this.editMode = false;
+        } else {
+          this.selectedCourse = null;
+          this.courseSelectorKey += 1; // courseSelectorKey counts the number of times the courseSelector has been open
```

Comment:
Not sure if this is the best pattern -- why is a key necessary for the component?

Reply:
Now that I'm looking at it, I don't think it is